### PR TITLE
Added Core concepts heading to Understanding odo

### PR DESCRIPTION
**Pull request by yhontyk**
- branches 4.2 and 4.3
- reviewed by the SMEs
- cc @bergerhoffer

### CODE REVIEW

File: cli_reference/openshift_developer_cli/understanding-odo.adoc [hunk 1]
```diff
@@ -20,4 +20,19 @@ Existing tools such as `oc` are more operations-focused and require a deep under
...
 * Partial compatibility with many languages and frameworks such as Ruby, Perl, PHP, and Python. Run odo catalog list components to see the list of supported and unsupported components.
 * Detects changes to local code and deploys it to the cluster automatically, giving instant feedback to validate changes in real time.
-* Lists all the available components and services from the {product-title} cluster.
\ No newline at end of file
+* Lists all the available components and services from the {product-title} cluster.
+
+== Core concepts
+
+Project::
+Project is your source code, tests, libraries organized in a separate single unit.
```
Comment (Preeticp) [inline cli_reference/openshift_developer_cli/understanding-odo.adoc]: tests, and libraries?

File: cli_reference/openshift_developer_cli/understanding-odo.adoc [hunk 2]
```diff
@@ -20,4 +20,19 @@ Existing tools such as `oc` are more operations-focused and require a deep under
 * Official full support of Node.js and Java components.
 * Partial compatibility with many languages and frameworks such as Ruby, Perl, PHP, and Python. Run odo catalog list components to see the list of supported and unsupported components.
 * Detects changes to local code and deploys it to the cluster automatically, giving instant feedback to validate changes in real time.
-* Lists all the available components and services from the {product-title} cluster.
\ No newline at end of file
+* Lists all the available components and services from the {product-title} cluster.
+
+== Core concepts
+
+Project::
```
Comment (Preeticp) [inline cli_reference/openshift_developer_cli/understanding-odo.adoc]: @boczkowska please add articles to the beginning of all these definitions: A project comprises your...; An application...; A component...; A service

File: cli_reference/openshift_developer_cli/understanding-odo.adoc [hunk 3]
```diff
@@ -20,4 +20,19 @@ Existing tools such as `oc` are more operations-focused and require a deep under
...
+Application::
+Application is a program designed for end users. An application consists of multiple microservices or components that work individually to build the entire application.
+Examples of applications: a video game, a media player, a web browser.
+Component::
+Component is a building block of an application. Typically it is a software package, a web service, a web resource, or a module that encapsulates a set of related functions or data. 
+Examples of components: Node.js, Pearl, PHP, Python, Ruby.
+Service::
+Service is a software that your component links to or depends on. 
+Examples of services: MariaDB, Jenkins, MySQL. 
+In {odo-title} services come from the OpenShift Service Catalog and must be enabled within your cluster.
```
Comment (Preeticp) [inline cli_reference/openshift_developer_cli/understanding-odo.adoc]: Comma after {odo-title}
s/come/are provisioned from

File: cli_reference/openshift_developer_cli/understanding-odo.adoc (hunk 2 above)
Comment (boczkowska) [reply cli_reference/openshift_developer_cli/understanding-odo.adoc]: Changed in https://github.com/openshift/openshift-docs/pull/18477/commits/5d462b2f0bd071a4900134053b985c98836d119d

**Review by bergerhoffer — COMMENTED** [review]
Just one minor thing, otherwise LGTM!

**Comment by boczkowska** [comment]
@bergerhoffer fixed! Thank you.

**Comment by bergerhoffer** [comment]
@boczkowska LGTM! Is this ready to be merged? If so, can you squash the commits? Thanks!

**Comment by boczkowska** [comment]
@bergerhoffer done! I do have a merge commit from rebasing, had a merge conflict and resolved it via UI. I hope it's okay.

**Pull request merged**